server/src/main/java/io/crate/analyze/where/DocKeys.java

```diff
@@ -63,31 +63,31 @@ public String getId(TransactionContext txnCtx, Functions functions, Row params,
             return idFunction.apply(
                 LazyMapList.of(
                     key.subList(0, width),
-                    s -> DataTypes.STRING.value(SymbolEvaluator.evaluate(txnCtx, functions, s, params, subQueryResults))
+                    s -> DataTypes.STRING.sanitizeValue(SymbolEvaluator.evaluate(txnCtx, functions, s, params, subQueryResults))
```

Comment:
Not sure if this is safe or whether we need the implicit cast here. Does this work for cases where a primary key column is a `bigint` or something like that?

Reply:
it was working before coz of https://github.com/crate/crate/pull/10226/commits/b27d1dc8183dd3e4feb0a70b0e324d1d494c9679#diff-d9273c5d03d846ae406a74ae5af32ad5L196 i dropped the `.toString` now.